### DISCUSSION TITLE
Link to specific Nested Tab

### DISCUSSION THREAD

**Discussion by NickGaultney** [elementor/elementor#21158] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

The new nested tabs are awesome, but I would like to link to a tab in the menu for easy & quick navigation.

### Describe the solution you'd like

Adding an id to each tab can allow you to jump to a specific tab much like snap scroll using "#my-section" to link to an individual section.

### Describe alternatives you've considered

Crocoblock JetElements

**Comment by coenvt** [comment]
Hi Nick, 

I was looking around and found this > https://code.colostate.edu/2021/04/01/opening-up-a-specific-tab-toggle-accordion-using-elementor/

You should adjust the CSS selectors (because of the new nested functionality): 

```
 // fetch all the tab, accordion, and toggle titles
 let desktoptitles = $('.elementor-tab-desktop-title');
 let mobiletitles = $('.elementor-tab-mobile-title');
```
to this:
```
 // fetch all the tab, accordion, and toggle titles
 let desktoptitles = $('.e-n-tabs-heading > .e-n-tab-title');
 let mobiletitles = $('.e-n-tabs-content > .e-n-tab-title');
```

And this:
 ```
// this check is needed for the tab module to work
 if (!$(desktoptitles).length)
 desktoptitles = $('.elementor-tab-title');
```

To this:
 ```
// this check is needed for the tab module to work
 if (!$(desktoptitles).length)
 desktoptitles = $('.e-n-tabs-heading > .e-n-tab-title');

```
If you copy this code with the adjusted css selectors mentioned above you can add it to the specific page with the 'html' widget from elementor. You should place the full code within <script> the code with new selectors </script> script tags.

Now you can add #yourtabname to your URL and directly link to that tab, it will open automaticly.

Regards, Coen (thanks to Shaun Geisert!) 

**If someone can assist to make this work on mobile please update the code and reply :-)**

**Comment by KHcreative** [comment]
For those of us not versed enough in JS to use it safely (meaning to make sure there are no potential security problems now or later), all that's left is to wait for Elementor to add this (in my opinion, basic) feature.
Thank you for making that request.

**Comment by Camilo517** [comment]
Fixed, it works on mobile and with svg. In my case I have fixed all the errors that I have found
```
add_action( 'wp_head', function () { ?>
    <script>
        (function ($) {
            document.addEventListener('DOMContentLoaded', function () {
                // fetch all the desktop tab, accordion, and toggle titles
                let desktoptitles = $('.e-n-tabs-heading > .e-n-tab-title');
                // this check is needed for the tab module to work
                if (!$(desktoptitles).length) desktoptitles = $('.e-n-tabs-heading > .e-n-tab-title');

                // fetch all the mobile tab, accordion, and toggle titles
                let mobiletitles = $('.e-n-tabs-content > .e-n-tab-title');
                // this check is needed for the tab module to work
                if (!$(mobiletitles).length) mobiletitles = $('.e-n-tabs-heading > .e-n-tab-title');

                function handleTitleClick(title) {
                    // clean up the titles to remove all non-alphanumeric chars, and prepend with an anchor
                    title = $(title).text().replace(/\W/g, '');
                    const regex = /[A-Z]+/;
                    title = title.match(regex).join('');

                    title = '#' + title;

                    var current_location = window.location.href;
                    current_location = current_location.split("#");

                    window.location = current_location[0] + title;
                }

                function handleClick(index, titles, offset) {
                    setTimeout(function () {
                        // trigger a click on the tab title
                        titles.eq(index).click();

                        // scroll to the correct position on the page.  Feel free to adjust offset
                        $('html, body').animate({
                            scrollTop: titles.eq(index).offset().top - offset
                        }, 'slow');

                    }, 1200);
                }

                // iterate the desktop titles
                $(desktoptitles).each(function (index, title) {
                    // compare the hash with the tab title
                    if (window.location.href.indexOf('#' + $(this).text().replace(/\W/g, '')) > -1) {
                        handleClick(index, desktoptitles, 250);
                    }
                });

                // also handle when users click desktop tabs (update links in the browser)
                $(desktoptitles).click(function (index, title) {
                    handleTitleClick(this);
                });

                // iterate the mobile titles
                $(mobiletitles).each(function (index, title) {
                    // compare the hash with the tab title
                    if (window.location.href.indexOf('#' + $(this).text().replace(/\W/g, '')) > -1) {
                        handleClick(index, mobiletitles, 0);
                    }
                });

                // also handle when users click mobile tabs (update links in the browser)
                $(mobiletitles).click(function (index, title) {
                    handleTitleClick(this);
                });

            });
        })(jQuery);
    </script>
<?php } );
```

**Reply by coenvt** [reply]
thanks for the update ! Camilo517 :-)

I'm going to test how it will work on my site.

**Reply by lensbreak** [reply]
@VerdiH Why is such simple improvement (really just a few lines of code) not yet part of elementor? Almost only need to copy/paste and would make elementor so much more usable. It's always those small little finishing-details that elementor lacks... And your customers/community even provide it to you, I do not understand why we have to wait for these things then still month after month. Come on!!

**Comment by drewclifton** [comment]
Yes I assumed this was a thing already and was surprised to find that it wasn't. The Tabs widget has been around for so long, yet doesn't have this fairly useful capability.

In my case, I would like to have buttons on the page that open a specific tab, as well as within another tab, that link to another tab (interlinking tabbed content).

**Comment by drewclifton** [comment]
I wanted to add more details about our use case:

There are cases when as designers we may wish to create a full-fledged _navigational_ experience, _without_ having to create a bunch of pages in WordPress and design them separately, as well as make users load pages individually (slow and not necessary).

In this context, the Tabs widget becomes like it's own set of mini pages with a navigation menu. Being able to interlink the tabbed content ("pages") becomes important.

Lastly, we need the ability to be able to optionally suppress the snapping effect that using anchor links creates. If a user clicks a link to open tab content, this should not (necessarily) impact the scroll depth of the page. In our layout (attached), we don't want snapping. Again, we want this to act more like a navigation menu loading several pages.

<img width="1321" alt="Elementor-Tabs-Widget-Deep-Links-Notes" src="https://github.com/elementor/elementor/assets/33555978/1192faa6-96b4-47c9-8e88-b6aae3409e4c">
This is a very convenient design method for both designers and visitors.
